File: app/src/main/java/com/ifpe/edu/br/model/model/auth/AirPowerUser.java
package com.ifpe.edu.br.model.model.auth;
/*
 * Trabalho de conclusão de curso - IFPE 2025
 * Author: Willian Santos
 * Project: AirPower Costumer
 */

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "AIR_POWER_USER")
public class AirPowerUser {

    @PrimaryKey
    @NonNull
    @ColumnInfo(name = "USER_ID")
    private String id;
    @ColumnInfo(name = "USER_AUTHORITY")
    private String authority;
    @ColumnInfo(name = "USER_COSTUMER_ID")
    private String customerId;
    @ColumnInfo(name = "USER_FIRST_NAME")
    private String firstName;
    @ColumnInfo(name = "USER_LAST_NAME")
    private String lastName;
    @ColumnInfo(name = "USER_NAME")
    private String name;
    @ColumnInfo(name = "USER_PHONE")
    private String phone;
    @ColumnInfo(name = "USER_EMAIL")
    private String email;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getAuthority() {
        return authority;
    }

    public void setAuthority(String authority) {
        this.authority = authority;
    }

    public String getCustomerId() {
        return customerId;
    }

    public void setCustomerId(String customerId) {
        this.customerId = customerId;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    @Override
    public String toString() {
        return "AirPowerUser{" +
                "id=" + id +
                ", authority='" + authority + '\'' +
                ", customerId='" + customerId + '\'' +
                ", firstName='" + firstName + '\'' +
                ", lastName='" + lastName + '\'' +
                ", name='" + name + '\'' +
                ", phone='" + phone + '\'' +
                ", email='" + email + '\'' +
                '}';
    }
}

File: app/src/main/java/com/ifpe/edu/br/model/repository/persistence/model/AirPowerUser.kt
package com.ifpe.edu.br.model.repository.persistence.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.ifpe.edu.br.model.repository.remote.dto.Id
import com.ifpe.edu.br.model.repository.remote.dto.user.ThingsBoardUser
import java.util.UUID

/*
* Trabalho de conclusão de curso - IFPE 2025
* Author: Willian Santos
* Project: AirPower Costumer
*/
@Entity(tableName = "AIR_POWER_USER")
data class AirPowerUser(
    @PrimaryKey
    @ColumnInfo(name = "USER_ID")
    val id: String,
    @ColumnInfo(name = "USER_AUTHORITY")
    val authority: String = "",
    @ColumnInfo(name = "USER_COSTUMER_ID")
    val customerId: String = "",
    @ColumnInfo(name = "TENANT_ID")
    val tenantId: String = "",
    @ColumnInfo(name = "USER_FIRST_NAME")
    val firstName: String? = "",
    @ColumnInfo(name = "USER_LAST_NAME")
    val lastName: String? = "",
    @ColumnInfo(name = "USER_NAME")
    var name: String? = "",
    @ColumnInfo(name = "USER_PHONE")
    val phone: String? = "",
    @ColumnInfo(name = "USER_EMAIL")
    val email: String = ""
) {
    override fun toString(): String {
        return "AirPowerUser(id='$id', authority='$authority', customerId='$customerId', tenantId='$tenantId', firstName=$firstName, lastName=$lastName, name=$name, phone=$phone, email='$email')"
    }
}

fun AirPowerUser.toThingsBoardUser(): ThingsBoardUser {
    return ThingsBoardUser(
        id = Id(UUID.fromString(id), "id"),
        createdTime = System.currentTimeMillis(),
        tenantId = Id(UUID.fromString(tenantId), "tenantId"),
        customerId = Id(UUID.fromString(customerId), "customerId"),
        email = email,
        name = name ?: "",
        authority = authority,
        firstName = firstName ?: "",
        lastName = lastName ?: "",
        phone = phone ?: "",
        additionalInfo = mapOf()
    )
}